share/doc/wake/logging.md

```diff
@@ -0,0 +1,111 @@
+# Logging & Console Output
+
+This article gives a high level overview of how wake logging works,
+and other things that impact what you see on the console when wake is executing or running database commands.
+Check out [print.wake](https://github.com/sifive/wake/blob/master/share/wake/lib/core/print.wake)
+for commented code.
+
+## Loggers
+
+Wake has a concept of loggers,
+to which bytes can be written during wake execution by various sources.
+Based on command line flags, each logger's output will be directed to
+zero or more of `wake`'s own output streams, possibly with additional formatting applied.
+The wake codebase and standard libraries often refer to a _logger_ as a `logLevel`.
+
+What's the difference between the different loggers?
+Which of wake's own output streams the results are sent to.
+This can influence what you see on your console when you run wake.
+The logger, in combination with the command line flags you invoke wake with,
+controls *what* you see. The logger also controls what it looks like (what color it is).
+It also controls which *actual* stream it is going to coming out of the executing `wake` process
+(stderr or stdout, or other file descriptors 3-5).
+
+`wake` has a number of built in loggers in wake which all start with `log...`.
+You can also make your own logger using `mkLogLevel`.
+You can use this to customize the color you want it to be shown as (if at all),
+and giving more fine-grained control over which wake output stream it should go to.
```

Comment:
```suggestion
You can use this to customize the color you want it to be shown as (if at all)
and give more fine-grained control over which wake output stream it should go to.
```

Grammatically, "customize" and "give" should be the same tense. Also, the comma should be removed because there's no subject in the second line.